### PR TITLE
Update meta.yaml and workflow to fetch git tag on master

### LINKED ISSUE #946: Make sure version 0.7.1 is released to conda
### What is your issue?

Conda runner failed when releasing v0.7.1

https://github.com/openghg/openghg/actions/runs/8114727980/job/22181909693

```
...

On branch devel
Your branch is up to date with 'origin/devel'.

nothing to commit, working tree clean

Updating build index: /home/runner/work/openghg/openghg/build

checkout: 'HEAD'
==> /usr/bin/git log -n1 <==

==> /usr/bin/git describe --tags --dirty <==

==> /usr/bin/git status <==

Adding in variants from internal_defaults
Adding in variants from /home/runner/work/openghg/openghg/recipes/conda_build_config.yaml
Attempting to finalize metadata for openghg
Error: Failed to render jinja template in /home/runner/work/openghg/openghg/recipes/meta.yaml:
'GIT_DESCRIBE_TAG' is undefined
Error: Process completed with exit code 1.
```

As much as was possible, the process within https://github.com/openghg/openghg/blob/14b8203826dec55a46105d5096078683cba8560c/doc/source/packaging.rst was followed for making the release.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)
     The value for the git tag is now fetched from the environment and passed into the conda recipe for the conda build.
     This ensures that the conda build on the master branch works as expected when the master is tagged instead of devel.

* **Please check if the PR fulfills these requirements**

- [x] Closes #946  (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`
- [x] Documentation and tutorials updated/added

### CODE REVIEW

File: doc/source/packaging.rst [hunk 1]
```diff
@@ -132,7 +87,7 @@ the command would be;
 
 .. code-block:: bash
 
-   git tag -a 0.12.0 -m "0.12.0 release"
+   git tag -a 0.7.1 -m "0.7.1 release"
```
Comment (rt17603) [inline doc/source/packaging.rst:90]: Our current tag messages are on the lines of - "OpenGHG release v0.7.1" so we should make we stick to that form for this. Checked using `$git tag -n`

```suggestion
   git tag -a 0.7.1 -m "OpenGHG release v0.7.1"
```

File: doc/source/packaging.rst [hunk 2]
```diff
@@ -147,63 +102,4 @@ that is included in the package to tag versions.
...
-
-replacing ``{VERSION}`` with the version number, e.g. for 0.11.2
-
-.. code-block:: bash
-
-   pip install metawards==0.11.2
-
-Play with the code, run the tests and run some examples. Everything should
-work as you have performed lots of prior testing to get to this stage.
+"main" branch.
```
Comment (rt17603) [inline doc/source/packaging.rst:105]:
```suggestion
"master" branch.
```

**Review by rt17603 — COMMENTED** [review]
Instructions look good to me - I've added a few quick comments. Are you happy this is the correct workflow now for completing a release?

For the updates to `workflow.yaml` and `meta.yaml` I'll look into this a bit more before final review.

**Comment by SutarPrasad** [comment]
Thanks Matt for your detail review and suggestions. As of changes required in current PR, I have updated those. And generated a separate issues to apply for enhanced workflow.

File: recipes/meta.yaml [hunk 3]
```diff
@@ -10,13 +10,15 @@ package:
 
 source:
   git_url: https://github.com/openghg/openghg.git
-  git_rev: {{ GIT_DESCRIBE_TAG }}
+  git_rev: {{ GIT_TAG }}
   git_depth: 1
```
Comment (milliams) [inline recipes/meta.yaml:14]: I _think_ that in order for the `GIT_DESCRIBE_TAG` bit in the `version` section to work, the depth here needs to be set to more than `1`. I'd just remove this `git_depth: 1` line as it's not required field and the repo is not so large that it should cause a problem.

**Review by milliams — CHANGES_REQUESTED** [review]

File: recipes/meta.yaml [hunk 4]
```diff
@@ -6,17 +6,19 @@
 
 package:
   name: "openghg"
-  version: {{ GIT_DESCRIBE_TAG }}
```
[the remark this answers is not in the document]
Comment (SutarPrasad) [reply recipes/meta.yaml]: updated back to GIT_DESCRIBE_TAG. And removed git_depth:1

File: recipes/meta.yaml (hunk 3 above)
Comment (SutarPrasad) [reply recipes/meta.yaml]: Its removed and pushed to PR.

File: recipes/meta.yaml [hunk 5]
```diff
@@ -10,13 +10,14 @@ package:
 
 source:
   git_url: https://github.com/openghg/openghg.git
-  git_rev: {{ GIT_DESCRIBE_TAG }}
-  git_depth: 1
+  git_rev: {{ GIT_TAG }}
```
Comment (milliams) [inline recipes/meta.yaml:13]:
```suggestion
  git_rev: {{ GIT_DESCRIBE_TAG }}
```
You were right to remove the `git_depth: 1` line, but I think this should still be `GIT_DESCRIBE_TAG` for the reasons I detailed before. Now that the Git checkout is not limited to one reference, the tags will be present and so `git describe` will work and so `GIT_DESCRIBE_TAG` should contain the correct value.

**Review by milliams — APPROVED** [review]

**Pull request merged**